### DISCUSSION TITLE
c++ API, fixed batch size for predictor.

### DISCUSSION THREAD

**Discussion by ashkanaev** [apache/incubator-mxnet#10937] [Q&A]
Hello!
Is't possible to make predictor with c++ API for arbitrary batch size? 
From  https://github.com/apache/incubator-mxnet/blob/97570916f844bcb4515d972c75fb0a75da345d97/example/image-classification/predict-cpp/image-classification-predict.cc

` const mx_uint input_shape_data[4] = { 1,
                                        static_cast<mx_uint>(channels),
                                        static_cast<mx_uint>(height),
                                        static_cast<mx_uint>(width) };
`

`MXPredCreate(static_cast<const char*>(json_data.GetBuffer()),
               static_cast<const char*>(param_data.GetBuffer()),
               static_cast<int>(param_data.GetLength()),
               dev_type,
               dev_id,
               num_input_nodes,
               input_keys,
               input_shape_indptr,
               input_shape_data,
               &pred_hnd);`

look's like predictor hard linked to fixed input data shape. how to make it flexible?

**Comment by vdantu** [comment]
@ashkanaev : You could post this question on discuss.mxnet.io to get a wider audience.  Also, are you seeing errors when you modify the predictor or is it not adhering to your modified input shapes? It would be helpful to know what you mean by making it flexible. 

@sandeep-krishnamurthy : Could you tag this with "C++", "Example" and "Question".

**Comment by chinakook** [comment]
MXPredReshape
However, reshape may bring overhead.

**Comment by RedbackThomson** [comment]
I am calling MXPredReshape with an updated input_shape_data (only changing the batch size), but when I do, the MXPredSetInput tells me my data size is larger than the expected shape. I am using the new handle, and even freeing the old handle. Can someone check to see if this is a bug in c_predict_api.cc or if I'm somehow messing it up.

**Comment by lupesko** [comment]
@ashkanaev thank you for submitting this issue!
This is an issue with the MXNet C Predict API. Let's please update the label to "C API" and remove the "C++" label.

@nswamy can you please update the labels?

**Comment by ThomasDelteil** [comment]
@leleamol could you pick this up ? Thanks!

**Comment by ou525** [comment]
@ashkanaev  Is this problem solved? How to achieve it?
